src/utils.c

```diff
@@ -7,6 +7,19 @@
 #include <stdlib.h>
 #include <string.h>
 
+char* string_cut_maxlen(char* string, int maxlen){
+        char *buffer;
+
+        if (maxlen < 0 || strlen(string) < maxlen) {
+                buffer = string;
+        } else {
+                size_t buffer_maxlen = maxlen;
+                buffer = calloc(buffer_maxlen, sizeof(char));
+                strncpy(buffer, string, buffer_maxlen);
```

Comment:
Quoting from `strcpy(3)`
> If there is no terminating null byte in the first n bytes of src, strncpy() produces an unterminated string in dest.

In other words at this point `buffer` is unterminated. The last character should be replaced with a null byte.